### PR TITLE
LVRUBYM-242:Create MyDayReport component

### PR DESCRIPTION
**Pull request by andriyarterchuk**
## JIRA

* [Main JIRA ticket](http://ssu-jira.softserveinc.com/browse/LVRUBYM-242)
* [JIRA subtask](http://ssu-jira.softserveinc.com/browse/LVRUBYM-242)

## Code reviewers

- [ ] @vikabelous
- [ ] @HelenMaherovska
- [ ] @andriyarterchuk

### Second Level Review

- [ ] @4asof4ik

## Summary of issue

Need to implement MyDayReport component

## Summary of change

- implement MyDayReport component

## Testing approach

Test with JSCS

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/js/myDayReport/components/myDayReport.component.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  });
+
+MyDayReportController.$inject = ['MyDayReport'];
+function MyDayReportController(MyDayReport) {
+  var ctrl = this;
+  ctrl.students = [];
+  MyDayReport.getReports().then(
+    function(data) {
+      ctrl.students = data;
+    }
```
Comment (4asof4ik) [inline app/js/myDayReport/components/myDayReport.component.js]: what if request will be no success??

File: app/js/myDayReport/components/myDayReport.component.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    );
+
+  ctrl.myDayReportUpdate = function(student) {
+    MyDayReport.updateReports(student.my_day_report.note, student.my_day_report.id)
+      .then(function(note) {
+      return note;
+    });
```
Comment (4asof4ik) [inline app/js/myDayReport/components/myDayReport.component.js]: line 19

File: app/js/myDayReport/myDayReport.route.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module.exports = angular
+  .module('myDayReport.route', [])
+  .config(function($stateProvider) {
+    $stateProvider.state({
+      url: '/my-day-report',
+      name: 'main.myDayReport',
```
Comment (4asof4ik) [inline app/js/myDayReport/myDayReport.route.js]: code style

File: test/spec/myDayReport.service.spec.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  afterEach(function() {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('is define', function() {
+    expect(service.getReports).toBeDefined();
+  });
+});
```
Comment (4asof4ik) [inline test/spec/myDayReport.service.spec.js]: Need more tests

File: app/js/common/services/my_day_report.service.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  var service = {
+    getReports: getReports,
+    updateReports: updateReports
+  };
+  return service;
+
+  function getReports() {
+    return myDayReportResource.query({group_id: currentGroupDay.group_id,
+                                      day: currentGroupDay.day}).$promise.then(function(data) {
+      return data;
```
Comment (4asof4ik) [inline app/js/common/services/my_day_report.service.js]: looks very ugly

File: app/js/common/services/my_day_report.service.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      return errors;
+    });
+  };
+
+  function updateReports(note, id) {
+    return myDayReportResource.update({report: {note: note},
+                                      id: id,
+                                      group_id: currentGroupDay.group_id,
+                                      day: currentGroupDay.day}).$promise.then(function(note) {
+      return note;
```
Comment (4asof4ik) [inline app/js/common/services/my_day_report.service.js]: looks very ugly

File: app/js/common/services/my_day_report.service.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    }, function(errors) {
+      alert(errors.data.errors);
```
Comment (4asof4ik) [inline app/js/common/services/my_day_report.service.js]: alert? bad idea

File: package.json [hunk 8]
```diff
@@ -7,6 +7,7 @@
     "server": "webpack-dev-server --history-api-fallback --watch-content-base --hot --progress --port 1111",
     "start": "npm run server",
     "deploy": "NODE_ENV=production webpack -p && NODE_ENV=production node server",
+    "run": "webpack -p",
```
Comment (4asof4ik) [inline package.json]: ????

File: test/spec/myDayReport.service.spec.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+  it('GetReports is define', function() {
+    expect(service.getReports).toBeDefined();
+  });
+
+  it('UpdateReports is define', function() {
+    expect(service.updateReports).toBeDefined();
+  });
+
+});
```
Comment (4asof4ik) [inline test/spec/myDayReport.service.spec.js]: add more tests

File: app/css/style.scss [hunk 10]
```diff
@@ -1571,3 +1571,7 @@ md-toast {
     margin: 0;
   }
 }
+
+.my-day .student-info {
+  width: 30%
+}
```
Comment (skrypnyk88) [inline app/css/style.scss]: Add empty line

File: app/js/common/resources/my_day_report.resource.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+var constants = require('./../constants');
+module.exports = angular
+  .module('myDayReport.resource', [
+    constants.name
+  ])
+  .factory('myDayReportResource', myDayReportResource);
+
+myDayReportResource.$inject = ['$resource', 'globalSettings', 'currentGroupDay'];
```
Comment (skrypnyk88) [inline app/js/common/resources/my_day_report.resource.js]: Do you need this `'currentGroupDay'`

File: app/js/common/services/my_day_report.service.js [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+module.exports = angular
+  .module('myDayReport.service', [
+    Resource.name,
+    toggleMessage.name
+    ])
+  .factory('MyDayReport', MyDayReport);
+
+MyDayReport.$inject = ['myDayReportResource',
+                       'currentGroupDay',
+                       'errorMessages',
```
Comment (skrypnyk88) [inline app/js/common/services/my_day_report.service.js]: Do you need this `'errorMessages',`

File: app/js/common/services/my_day_report.service.js [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+                       'toggleMessage'];
+
+function MyDayReport(myDayReportResource,
+                     currentGroupDay,
+                     errorMessages,
```
Comment (skrypnyk88) [inline app/js/common/services/my_day_report.service.js]: Same here

File: app/js/common/services/my_day_report.service.js [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+                     toggleMessage) {
+  var service = {
+    getReports: getReports,
+    updateReports: updateReports
+  };
+  return service;
+
+  function getReports() {
+    var params = {group_id: currentGroupDay.group_id};
+    return myDayReportResource.query(params)
```
Comment (skrypnyk88) [inline app/js/common/services/my_day_report.service.js]: Wright `.$promise` in this line

File: app/js/common/services/my_day_report.service.js [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+    .$promise
+    .then(responseSuccess, responseFailure);
+  };
+
+  function updateReports(note, id) {
+    var params = {report: {note: note},
+                  id: id,
+                  group_id: currentGroupDay.group_id
+                };
```
Comment (skrypnyk88) [inline app/js/common/services/my_day_report.service.js]: +    var params = {report: {note: note},
 +                          id: id,
 +                          group_id: currentGroupDay.group_id
 +                          };

File: app/js/common/services/my_day_report.service.js [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+    return myDayReportResource.update(params)
```
Comment (skrypnyk88) [inline app/js/common/services/my_day_report.service.js]: Wright  `.$promise`  in this line

File: app/js/main/main.module.js [hunk 17]
```diff
@@ -31,7 +32,9 @@ module.exports = angular
     profileModule.name,
     forgotPasswordModule.name,
     resetPasswordModule.name,
-    healthReportModule.name
+    healthReportModule.name,
+    staticModule.name,
```
Comment (skrypnyk88) [inline app/js/main/main.module.js]: Why?

File: app/js/myDayReport/components/myDayReport.component.js [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+  ])
+  .component('myDayReport', {
+    templateUrl: './app/js/myDayReport/components/myDayReport.template.html',
+    controller: MyDayReportController
+  });
+
+MyDayReportController.$inject = ['MyDayReport'];
+function MyDayReportController(MyDayReport) {
+  var ctrl = this;
+  ctrl.students = [];
```
Comment (skrypnyk88) [inline app/js/myDayReport/components/myDayReport.component.js]: ctrl.reports

File: app/js/myDayReport/components/myDayReport.component.js [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+  MyDayReport.getReports().then(
+    function(data) {
+      ctrl.students = data;
```
Comment (skrypnyk88) [inline app/js/myDayReport/components/myDayReport.component.js]: the same

File: app/js/myDayReport/components/myDayReport.template.html [hunk 20]
```diff
@@ -0,0 +1,11 @@
+<div class="report my-day" ng-repeat="report in $ctrl.students">
```
Comment (skrypnyk88) [inline app/js/myDayReport/components/myDayReport.template.html]: ctrl.reports

File: app/js/myDayReport/myDayReport.route.js [hunk 21]
```diff
@@ -0,0 +1,8 @@
+module.exports = angular
+  .module('myDayReport.route', [])
+  .config(function($stateProvider) {
```
Comment (skrypnyk88) [inline app/js/myDayReport/myDayReport.route.js]: please inject `$stateProvider`

File: app/js/myDayReport/components/myDayReport.template.html [hunk 22]
```diff
@@ -0,0 +1,11 @@
...
+  <md-list>
+    <md-list-item>
+      <student-component student="report.student" class="student-info" layout="row"></student-component>
+        <form ng-submit="$ctrl.myDayReportUpdate(report)">
```
Comment (yaremakleputs) [inline app/js/myDayReport/components/myDayReport.template.html]: It looks like form inside student component.

File: test/spec/myDayReport.service.spec.js [hunk 23]
```diff
@@ -0,0 +1,86 @@
...
+    spyOn(toggleMessage, 'showMessages');
+    service.updateReports(note, id).then(function(updatedReport) {
+      response = updatedReport;
+    });
+
+    $httpBackend.flush();
+
+    expect(JSON.stringify(response)).toEqual(JSON.stringify(report));
+  }));
+
```
Comment (yaremakleputs) [inline test/spec/myDayReport.service.spec.js]: Do you need this empty line?

**Pull request merged**